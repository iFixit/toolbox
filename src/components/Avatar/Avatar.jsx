import React from 'react';
import { oneOfType, string, number } from 'prop-types';
import glamorous from 'glamorous';

const AvatarContainer = glamorous.img(props => ({
   width: props.size,
   height: props.size,
   borderRadius: '50%',
}));

const Avatar = props => <AvatarContainer {...props} />;

Avatar.propTypes = {
   /** The avatar image URL. */
   src: string.isRequired,
   /**
    * The width and height of the avatar.
    * A number will be interpreted as a pixel value.
    * Use a string to specify other units (e.g. "2em").
    * */
   size: oneOfType([string, number]),
};

Avatar.defaultProps = {
   size: 48,
};

export default Avatar;
