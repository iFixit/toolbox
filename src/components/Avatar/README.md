### Examples

#### Default Avatar
```
<Avatar src="https://d3nevzfk7ii3be.cloudfront.net/igi/MrphvTGGRcKHXRV2.standard" />
```

#### Custom sizes
```
<Avatar
   src="https://d3nevzfk7ii3be.cloudfront.net/igi/kYipBhTkDKX4aXIY.standard"
   size={32}
/>
```

```
<Avatar
   src="https://d3nevzfk7ii3be.cloudfront.net/igi/kYipBhTkDKX4aXIY.standard"
   size="4rem"
/>
```
