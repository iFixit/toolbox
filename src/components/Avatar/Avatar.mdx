---
name: Avatar
route: /components/avatar
menu: Components
---

import Avatar from './Avatar'
import { Playground, PropsTable } from 'docz'

# Avatar

## Examples

### Default

<Playground>
  <Avatar src="https://d3nevzfk7ii3be.cloudfront.net/igi/MrphvTGGRcKHXRV2.standard" />
</Playground>

### Custom sizes

<Playground>
  <Avatar
    src="https://d3nevzfk7ii3be.cloudfront.net/igi/kYipBhTkDKX4aXIY.standard"
    size={32}
  />
</Playground>

<Playground>
  <Avatar
    src="https://d3nevzfk7ii3be.cloudfront.net/igi/kYipBhTkDKX4aXIY.standard"
    size="4em"
  />
</Playground>

## Props

<PropsTable of={Avatar} />
