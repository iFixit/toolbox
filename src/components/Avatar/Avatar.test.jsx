import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

test('renders without crashing', () => {
   const tree = renderer
      .create(<Avatar src="http://via.placeholder.com/100x100" />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom size (number)', () => {
   const tree = renderer
      .create(<Avatar src="http://via.placeholder.com/100x100" size={32} />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom size (string)', () => {
   const tree = renderer
      .create(<Avatar src="http://via.placeholder.com/100x100" size="4rem" />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});
