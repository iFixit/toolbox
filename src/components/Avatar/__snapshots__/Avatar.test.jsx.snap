// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom size (number) 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
}

<img
  className="c0"
  size={32}
  src="http://via.placeholder.com/100x100"
/>
`;

exports[`renders with custom size (string) 1`] = `
.c0 {
  width: 4rem;
  height: 4rem;
  border-radius: 50%;
}

<img
  className="c0"
  size="4rem"
  src="http://via.placeholder.com/100x100"
/>
`;

exports[`renders without crashing 1`] = `
.c0 {
  width: 48px;
  height: 48px;
  border-radius: 50%;
}

<img
  className="c0"
  size={48}
  src="http://via.placeholder.com/100x100"
/>
`;
